modules/ding_provider/ding_provider.module

```diff
@@ -612,144 +612,15 @@ function ding_provider_get_pseudo_entity($ding_entity_id) {
-    case -1:
-      if (preg_match('/^' . $local_range . '/', $id)) {
-        // If agency is not given fallback to local agency.
-        if (empty($agency) || $agency == -1) {
-          $agency = $local_agency;
-        }
-
-        // Local posts.
-        $ting_entity_id = $agency . '-katalog:' . $id;
-      }
-      else {
-        // Global posts.
-        $ting_entity_id = '870970-basis:' . $id;
-      }
-      break;
-
-    case '150021':
-      $ting_entity_id = '150021-bibliotek:' . $id;
-      break;
-
-    case '150027':
-      $ting_entity_id = '150021-fjern:' . $id;
-      break;
-
-    case '150030':
-      $ting_entity_id = '870970-spilmedier:' . $id;
-      break;
-
-    case '870973':
-      $ting_entity_id = '870973-anmeld:' . $id;
-      break;
-
-    case '150039':
-      $ting_entity_id = '150015-forlag:' . $id;
-      break;
-
-    case '870976':
-      $ting_entity_id = '870976-anmeld:' . $id;
-      break;
-
-    case '150048':
-      $ting_entity_id = '870970-basis:' . $id;
-      break;
-
-    case '150028':
-      $ting_entity_id = '870970-basis:' . $id;
-      break;
-
-    case '150015':
-      $ting_entity_id = '870970-basis:' . $id;
-      break;
-
-    case '150033':
-      $ting_entity_id = '150033-dandyr:' . $id;
-      break;
-
-    case '150040':
-      $ting_entity_id = '150040-verdyr:' . $id;
-      break;
-
-    case '150018':
-      $ting_entity_id = '150018-danhist:' . $id;
-      break;
-
-    case '150032':
-      $ting_entity_id = '150018-samfund:' . $id;
-      break;
-
-    case '150034':
-      $ting_entity_id = '150018-religion:' . $id;
-      break;
-
-    case '150054':
-      $ting_entity_id = '150018-biologi:' . $id;
-      break;
-
-    case '150055':
-      $ting_entity_id = '150018-fysikkemi:' . $id;
-      break;
-
-    case '150056':
-      $ting_entity_id = '150056-geografi:' . $id;
-      break;
-
-    case '150008':
-      $ting_entity_id = '150008-academic:' . $id;
-      break;
-
-    case '150043':
-      $ting_entity_id = '150043-atlas:' . $id;
-      break;
-
-    case '150023':
-      $ting_entity_id = '150023-sicref:' . $id;
-      break;
-
-    case '150025':
-      $ting_entity_id = '150008-public:' . $id;
-      break;
-
-    case '150052':
-      $ting_entity_id = '870970-basis:' . $id;
-      break;
-
-    case '159002':
-      $ting_entity_id = '159002-lokalbibl:' . $id;
-      break;
-
-    case '150012':
-      $ting_entity_id = '150012-leksikon:' . $id;
-      break;
-
-    default:
-      // The agency ID was not translatable, do to one-to-may match from old pid
-      // to new pid. Or the agency ID is not known. So simply log it under a
-      // filterable type.
-      watchdog('ding_provider_pid', 'PID creation failed with agency (%agency) and ID (%id)', array('%agency' => $agency, '%id' => $id), WATCHDOG_ERROR);
-      break;
-  }
+function ding_provider_build_entity_id($id) {
+  $obj = ting_do_search($id . ' AND facet.acSource=bibliotekskatalog');
```

Comment:
should be: ting_do_search('rec.id=' . $id . ' AND facet.acSource=bibliotekskatalog');